Patch the Makefile of libcap to work with the newest gperf 3.1, until
an upstream fix is available.


Signed-off-by: Clemens Gruber <clemens.gru...@pqgruber.com>
---
 .../0002-Make-compatible-with-gperf-3.1.patch      | 22 ++++++++++++++++++++++
 patches/libcap-2.25/series                         |  3 ++-
 2 files changed, 24 insertions(+), 1 deletion(-)
 create mode 100644 
patches/libcap-2.25/0002-Make-compatible-with-gperf-3.1.patch

diff --git a/patches/libcap-2.25/0002-Make-compatible-with-gperf-3.1.patch 
b/patches/libcap-2.25/0002-Make-compatible-with-gperf-3.1.patch
new file mode 100644
index 000000000..93ef47364
--- /dev/null
+++ b/patches/libcap-2.25/0002-Make-compatible-with-gperf-3.1.patch
@@ -0,0 +1,22 @@
+From: Clemens Gruber <clemens.gru...@pqgruber.com>
+Date: Fri, 31 Mar 2017 16:25:26 +0200
+Subject: [PATCH] Make compatible with gperf 3.1
+
+Signed-off-by: Clemens Gruber <clemens.gru...@pqgruber.com>
+---
+ libcap/Makefile | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/libcap/Makefile b/libcap/Makefile
+index d189777baa0f..b7939dd30c11 100644
+--- a/libcap/Makefile
++++ b/libcap/Makefile
+@@ -41,7 +41,7 @@ cap_names.h: _makenames
+       ./_makenames > cap_names.h
+ 
+ $(GPERF_OUTPUT): cap_names.list.h
+-      perl -e 'print "struct __cap_token_s { const char *name; int index; 
};\n%{\nconst struct __cap_token_s *__cap_lookup_name(const char *, unsigned 
int);\n%}\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; 
print $$l; }' < $< | gperf --ignore-case --language=ANSI-C --readonly 
--null-strings --global-table --hash-function-name=__cap_hash_name 
--lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
++      perl -e 'print "struct __cap_token_s { const char *name; int index; 
};\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; print $$l; 
}' < $< | gperf --ignore-case --language=ANSI-C --includes --readonly-tables 
--null-strings --global-table --hash-function-name=__cap_hash_name 
--lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
+ 
+ cap_names.list.h: Makefile $(KERNEL_HEADERS)/linux/capability.h
+       @echo "=> making $@ from $(KERNEL_HEADERS)/linux/capability.h"
diff --git a/patches/libcap-2.25/series b/patches/libcap-2.25/series
index 4f7f7421b..60633953b 100644
--- a/patches/libcap-2.25/series
+++ b/patches/libcap-2.25/series
@@ -1,4 +1,5 @@
 # generated by git-ptx-patches
 #tag:base --start-number 1
 0001-Fix-build-on-non-Linux-host.patch
-# 69b7a8db215659ad00eab9b8ed2ab43e  - git-ptx-patches magic
+0002-Make-compatible-with-gperf-3.1.patch
+# 84f6c460166806b31adde9353aa78a88  - git-ptx-patches magic
-- 
2.12.1


_______________________________________________
ptxdist mailing list
ptxdist@pengutronix.de

    Previous message
    View by thread
    View by date
    Next message

Reply via email to
The Mail Archive

    The Mail Archive home
    ptxdist - all messages
    ptxdist - about the list
    Expand
    Previous message
    Next message

